### DISCUSSION TITLE
New note filename custom formats

### DISCUSSION THREAD

**Discussion by YodaEmbedding** [GitJournal/GitJournal#685] [Feature Requests]
More customization for the format would be nice. Personally, I want titles lowercase with underscores:

```
This is title  ->  this_is_title.md
```

This is good since Linux paths and commands are all easier to use with lowercase and no spaces.

**Comment by vHanda** [comment]
Hi. I'll be happy to implement this, however, maybe it would be better to implement some kind of templating language and expose that.

Would you have some time to look into a simple templating language?

**Comment by YodaEmbedding** [comment]
Here's one proposal using `{{key:value}}` [style](https://help.obsidian.md/Plugins/Templates).

```
filename                        template

2021-12-24                      {{date}}
2021-12-24 23:59:59             {{date:YYYY-MM-DD hh:mm:ss}}
2021-12-24T23:59:59.999+0700    {{date:YYYY-MM-DD'T'hh:mm:ss.SSSZ}}
23:59:59                        {{time}}

Title text                      {{title}}
title_text                      {{title:lowercase,underscore}}
title_t                         {{title:lowercase,underscore,max_length=7}}
titl...                         {{title:lowercase,underscore,max_length=7,max_continue=...}}

76363445-2654-48...8d85         {{uuidv4}}
$ROOT/new_notes/1               {{root}}/new_notes/{{counter:start=1,next_from=last}}
```

I only need `title_text` but the others should be able to cover other use cases.

**Comment by justinsilvestre** [comment]
I couldn't find an existing package for template strings that looked suitable for this, but I managed to roll one myself that does the most basic parts. I could only test it in a Dart sandbox (as I couldn't figure out how to get the tests in this repo running) but maybe with some direction + free time I could write some tests.

https://github.com/GitJournal/GitJournal/compare/master...justinsilvestre:GitJournal:master

I don't have time to figure out the UI + Flutter bits, unfortunately. If someone else could step in to do those parts + tests, that would be much appreciated!

**Comment by justinsilvestre** [comment]
Here's where I tested that code linked above: https://dartpad.dev/?id=515ac6e065e5e6cb3fac83bff9de0f63
